x/tokenfactory/keeper/createdenom.go

```diff
@@ -81,6 +81,17 @@ func (k Keeper) chargeForCreateDenom(ctx sdk.Context, creatorAddr string, _ stri
 		if err := k.communityPoolKeeper.FundCommunityPool(ctx, creationFee, accAddr); err != nil {
 			return err
 		}
+	} else {
+		// charge a useless gas cost of 2 million here by setting the param repeatedly, then resetting it back to the default
+		defaultParams := k.GetParams(ctx)
+
+		tempParams := defaultParams
+		for i := 1; i < 500; i++ {
+			tempParams.DenomCreationFee = sdk.NewCoins(sdk.NewCoin(sdk.DefaultBondDenom, sdk.NewInt(int64(i))))
+			k.SetParams(ctx, tempParams)
+		}
```

Comment:
this is unnecessary - the same effect could be achieved with 
```go
ctx.GasMeter().ConsumeGas(storetypes.Gas(20000000), "consume denom creation gas")
```

Reply:
Sweet! Thank you this is what I was looking for 😄 I appreciate it